modules/apps/object/object-dynamic-data-mapping-form-field-type/src/main/java/com/liferay/object/dynamic/data/mapping/form/field/type/internal/attachment/AttachmentDDMFormFieldTemplateContextContributor.java

```diff
@@ -84,13 +84,19 @@ public Map<String, Object> getParameters(
 			_uploadServletRequestConfigurationHelper.getMaxSize()
 		).put(
 			"tip",
-			_language.format(
-				ddmFormFieldRenderingContext.getLocale(),
-				"upload-a-x-no-larger-than-x-mb",
-				new Object[] {
-					ddmFormField.getProperty("acceptedFileExtensions"),
-					maximumFileSize
-				})
+			() -> {
+				if ((boolean)ddmFormField.getProperty("readOnly")) {
+					return null;
+				}
+
+				return _language.format(
+					ddmFormFieldRenderingContext.getLocale(),
+					"upload-a-x-no-larger-than-x-mb",
+					new Object[] {
+						ddmFormField.getProperty("acceptedFileExtensions"),
+						maximumFileSize
+					});
+			}
```

Comment:
Hey @Manda-vit, I don't think we need to do this validation on the backend, if you see that this `tip` property is being passed to the frontend, then I think we'd better get it on the frontend and do the validation there, dropping this backend validation.

1. For that you can add a new `tip` prop of type `string` in `IProps` to `Attachment.tsx` and in the Attachment component itself you can already have access to this property in the frontend.

2. The component responsible for showing this `tip` message is `FieldBase`, so we make the condition in it, like:

```tsx
<FieldBase
	readOnly={readOnly}
	tip={!readOnly ? tip : ""} 
	{...otherProps} 
	{...error}
>
```